docs/en/Configuring-Skill-Instance.rst

```diff
@@ -0,0 +1,88 @@
+
+The ``SkillBuilder`` provides helper functions for constructing the ``Skill``, setting custom user agent and create lambda integration handler. ASK SDK v2 for Node.js provides different implementation of ``SkillBuilder`` that offer different level of customization support, which is available through ``SkillBuilders`` object.
+
+The following example shows how to create skill builders using ``SkillBuilders`` provider.
+
+.. tabs::
+
+  .. code-tab:: javascript
+
+    const Alexa = require('ask-sdk');
+
+    const customSkillBuilder = Alexa.SkillBuilders.custom();
+    const standardSkillBuilder = Alexa.SkillBuilders.standard();
+
+  .. code-tab:: typescript
+
+    import { SkillBuilders } from 'ask-sdk';
+
+    const customSkillBuilder = SkillBuilders.custom();
+    const standardSkillBuilder = SkillBuilders.standard()
+
+
+BaseSkillBuilder
+----------------
+
+``BaseSkillBuilder`` includes the most essentially methods for configuring ``Skill``. It serves as the base for any future customization of the ``SkillBuilder``
+
+Available Methods
+^^^^^^^^^^^^^^^^^
+
+.. code-block:: typescript
+
+  addRequestHandler(matcher : ((handlerInput : HandlerInput) => Promise<boolean> | boolean) | string, executor : (handlerInput : HandlerInput) => Promise<Response> | Response) : this;
+  addRequestHandlers(...requestHandlers : RequestHandler[]) : this;
+  addRequestInterceptors(...executors : Array<RequestInterceptor | ((handlerInput : HandlerInput) => Promise<void> | void)>) : this;
+  addResponseInterceptors(...executors : Array<ResponseInterceptor | ((handlerInput : HandlerInput, response? : Response) => Promise<void> | void)>) : this;
+  addErrorHandler(matcher : (handlerInput : HandlerInput, error : Error) => Promise<boolean> | boolean, executor : (handlerInput : HandlerInput, error : Error) => Promise<Response> | Response) : this;
+  addErrorHandlers(...errorHandlers : ErrorHandler[]) : this;
+  withCustomUserAgent(customUserAgent : string) : this;
+  withSkillId(skillId : string) : this;
+  getSkillConfiguration() : SkillConfiguration;
+  create() : Skill;
+  lambda() : LambdaHandler;
+
+CustomSkillBuilder
+--------------------
+
+``CustomSkillBuilder`` is available in both ``ask-sdk-core`` and ``ask-sdk`` package and includes all available methods from ``BaseSkillBuilder``. In addition, ``CustomSkillBuilder`` also provides functions that allows you to register custom ``PersistentAdapter`` and ``ApiClient``.
+
+Available Methods
+^^^^^^^^^^^^^^^^^
+
+.. code-block:: typescript
+
+  withPersistenceAdapter(persistenceAdapter : PersistenceAdapter) : this;
+  withApiClient(apiClient : ApiClient) : this;
+
+Standard Skill Builder
+----------------------
+
+``StandardSkillBuilder`` is available only in the ``ask-sdk`` package and includes all available methods from ``BaseSkillBuilder``. It uses ``DynamoDbPersistenceAdapter`` and ``DefaultApiClient`` by default and provides conveninent helper methods for configuring ``DynamoDbPersistenceAdapter``.
```

Comment:
I guess it would be good to add where these ``DynamoDbPersistenceAdapter`` and ``DefaultApiClient`` classes live. This is as part of a doc feedback we received earlier.

Reply:
Those are called out in the corresponding pages. e.g.: [link](https://ask-sdk-for-nodejstest.readthedocs.io/en/latest/Managing-Attributes.html#dynamodbpersistenceadapter)